crates/metadata-providers/src/objstore/optimistic_store.rs

```diff
@@ -106,24 +151,29 @@ impl OptimisticLockingMetadataStore {
         value: VersionedValue,
         precondition: Precondition,
     ) -> Result<(), WriteError> {
-        // create a random cookie
-        let cookie = random::<u64>();
-        let buf = self.serialize_versioned_value(&value, cookie)?;
+        // create a random salt
+        // The salt is a random value to break CAS ties.
+        // S3 cas working by comparing ETags (hash of the body) and we don't want a case where two writer think they both succeeded the cas (they both CASed the same body)
```

Comment:
Minor cleanup for your consideration:

```suggestion
        // The salt is a random value to break CAS ties. S3 CAS works by comparing ETags (body hashes).
        // We want to avoid two writers thinking they both succeeded if they happen to write identical bodies.
```

Reply:
Done! :)